woocommerce/wc-api-dev
# Add a `counts` endpoint to the data endpoints

This PR adds a new endpoint at `/wc/v3/data/counts`, which does some tallying of products, orders, and reviews to give a one-stop-spot for counting up various site objects. This will be used in Calypso to handle conditionally showing widgets on the dashboard, showing "pending" counts on the sidebar; without having to do requests to each object's endpoint separately.

This does some caching for products & reviews, just for 15 minutes, to prevent repeated calculations. For orders, `wp_count_posts` is cached by core.

An example response:

```
{
	"orders": {
		"wc-pending": 0,
		"wc-processing": 1,
		"wc-on-hold": 0,
		"wc-completed": 1,
		"wc-cancelled": 0,
		"wc-refunded": 0,
		"wc-failed": 0
	},
	"products": {
		"all": 5,
		"low-inventory": 2,
		"out-of-stock": 1
	},
	"reviews": {
		"approved": 1,
		"awaiting_moderation": 1,
		"spam": 0,
		"trash": 0,
		"post-trashed": 0,
		"total_comments": 2,
		"all": 2
	}
}
```

To test:

- GET `/wc/v3/data/counts`, make sure the counts are correct
- Run the tests: `phpunit` (or `phpunit tests/unit-tests/data-counts.php`)
